Adding pre-push hook (#3857)
antrea-io/antrea: CONTRIBUTING.md

[starter](https://github.com/antrea-io/antrea/labels/Good%20first%20issue)
issues from which you can choose.

### Git Client Hooks

> There is a recommended git-client-hooks git hook which we advise you use. You can find
> it here:

Reviewer: ```suggestion
There are a few recommended git client hooks which we advise you to use. You can find
them here:
```

Reviewer 2: +1

[hack/git_client_side_hooks/pre-commit](hack/git_client_side_hooks/pre-commit).
You can run `make install-hooks` to copy it to your local `.git/hooks/` folder, and remove it via `make uninstall-hooks`

> ### Pre-Push Hook

Reviewer: I think you can use `Git Client Hooks` as the title to merge `Pre-Commit` and `Pre-Push` hooks.
